lib/pdf/core/xmp_metadata.rb

```diff
@@ -0,0 +1,114 @@
+require 'nokogiri'
+
+module PDF
+  module Core
+    class XmpMetadata
+      attr_accessor :enable_pdfa_1b
+
+      # These attributes must all be synchronized with their counterparts in the
+      # document information dictionary to be PDF/A-1b compliant.
+      attr_accessor :dc_title, :dc_creator, :dc_description,
+        :pdf_keywords, :xmp_creator_tool, :pdf_producer,
+        :xmp_create_date, :xmp_modify_date
+
+      def initialize(options = {})
+        @xml_doc = Nokogiri::XML(
+          "<rdf:RDF xmlns:rdf='http://www.w3.org/1999/02/22-rdf-syntax-ns#'/>"
+        )
+
+        # Convert options for the document information dictionary to their
+        # counterparts in XMP.
+        @dc_title = options[:Title] if options[:Title]
+        @dc_creator = options[:Author] if options[:Author]
+        @dc_description = options[:Subject] if options[:Subject]
+        @pdf_keywords = options[:Keywords] if options[:Keywords]
+        @xmp_creator_tool = options[:Creator] if options[:Creator]
+        @pdf_producer = options[:Producer] if options[:Producer]
+        @xmp_create_date = options[:CreationDate] if options[:CreationDate]
+        @xmp_modify_date = options[:ModDate] if options[:ModDate]
+      end
+
+      def render
+        render_pdfaid if @enable_pdfa_1b
+        render_xmp if @xmp_creator_tool || @xmp_create_date || @xmp_modify_date
+        render_pdf if @pdf_keywords || @pdf_producer
+        render_dc if @dc_title || @dc_creator || @dc_description
+        @xml_doc.root.to_xml
```

Comment:
I don't have to use nokogiri. It was my first shot at the problem because I use it elsewhere. I accept that no external dependencies should be added and will look into changing it to string interpolations.